drivers/ethernet/eth_stm32_hal.c

```diff
@@ -210,9 +209,11 @@ static int eth_tx(struct device *dev, struct net_pkt *pkt)
 	}
 
 #if defined(CONFIG_SOC_SERIES_STM32H7X)
-	ETH_BufferTypeDef tx_buffer_def[ETH_TXBUFNB];
+	/* Only one tx_buffer is sufficient to pass only 1 dma_buffer */
+	/* We can save some memory here by allocating only 1 buffer */
+	ETH_BufferTypeDef tx_buffer_def[1];
```

Comment:
I would use `ETH_TXBUFNB` define here and change it's value in code above to appropriate one. Please put this change into separate memory management commit.

Reply:
I have split into 2 commits, but ETH_TXBUFNB should be changed as they are used in the dma_tx_desc_tab. The ETH_BufferTypeDef from the HAL is just a linked list from which we can pass successive arrays to be transmitted. Here we only pass 1 dma_buffer corresponding to the first dma_buffer filled up with the net_buffer. So these symbols are differents.

Reply:
Thanks for splitting into 2 commits. Then introduce new one define with meaningful name (e.g. `ETH_TXBUF_DEF_LEN`) and value (1U in your case).

Reply:
I also thought of it in the sense that we only fill up 1 DMA descriptor at a time. 1 descriptor is used to transmit 1 ethernet frame. At the beginning of the function we exit if the size of the packet is bigger that the eth frame size. So this function needs a lot of work in order to send multiple frames at once. I will add the new symbol btw.

Reply:
@Nukersson  Changed to new define ETH_TXBUF_DEF_NB set to 1U.

Reply:
> I also thought of it in the sense that we only fill up 1 DMA descriptor at a time. 1 descriptor is used to transmit 1 ethernet frame. At the beginning of the function we exit if the size of the packet is bigger that the eth frame size. So this function needs a lot of work in order to send multiple frames at once. I will add the new symbol btw.

Same problem with budders I had also. I agree with you. We need only one tx buffer but not 4 as proposed in original driver. I think wi will fix it in this PR or open later a new one. As I said - I would test your PR later today.

Reply:
> At the beginning of the function we exit if the size of the packet is bigger that the eth frame size.

Note that upper layer in network stack will not send bigger packets that can fit into Ethernet frame.

Reply:
> > At the beginning of the function we exit if the size of the packet is bigger that the eth frame size.
> 
> Note that upper layer in network stack will not send bigger packets that can fit into Ethernet frame.

Is it KConfig define then? Can you tell the name of this define, if yes?

Reply:
> Is it KConfig define then? Can you tell the name of this define, if yes?

I do not understand what you mean by this. What defines are you looking for?

Reply:
> > Is it KConfig define then? Can you tell the name of this define, if yes?
> 
> I do not understand what you mean by this. What defines are you looking for?

I am looking for upper layer stack variable/define, which needed to configures packet's size for Ethernet frame.

Reply:
For that you can use `NET_ETH_MAX_FRAME_SIZE`

Reply:
> > > Is it KConfig define then? Can you tell the name of this define, if yes?
> > 
> > 
> > I do not understand what you mean by this. What defines are you looking for?
> 
> I am looking for upper layer stack variable/define, which needed to configures packet's size for Ethernet frame.

Thank you very much.

Reply:
Thank you very much!

Reply:
> Same problem with budders I had also. I agree with you. We need only one tx buffer but not 4 as proposed in original driver. I think wi will fix it in this PR or open later a new one. As I said - I would test your PR later today.

@Nukersson I suggest to add this improvement into a new PR as the IT approach is a minimal change. Changing the handling of the buffer needs more time and testing to get working properly (as it plays with the net stack itself). We'll also have to consider checking for the availability of every DMA TX Descriptors before initiating a TX call because HAL_TransmitIT could fail if one of the DMA descriptors is not available.

Reply:
Ok. Sounds solide.